### PR TITLE
python310Packages.markdown2: 2.4.8 -> 2.4.10

### PR DESCRIPTION
**Pull request by happysalada**
## Description of changes

minor update and add optional dependencies

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
@hbunke if you have time to review.

File: pkgs/development/python-modules/markdown2/default.nix [hunk 1]
```diff
@@ -3,18 +3,22 @@
...
+, pythonOlder
+, wavedrom
 }:
 
 buildPythonPackage rec {
   pname = "markdown2";
-  version = "2.4.8";
+  version = "2.4.10";
+
+  disable = pythonOlder "3.5";
```
Comment (dotlambda) [inline pkgs/development/python-modules/markdown2/default.nix:14]:
```suggestion
  disabled = pythonOlder "3.5";
```

File: pkgs/development/python-modules/markdown2/default.nix [hunk 2]
```diff
@@ -29,6 +33,16 @@ buildPythonPackage rec {
...
   '';
 
+  passthru.optional-dependencies = {
+    code_syntax_highlighting = [
+      pygments
+    ];
+    wavedrom = [
+      wavedrom
+    ];
+    all = [ pygments wavedrom ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/markdown2/default.nix:43]:
```suggestion
    all = lib.flatten (lib.attrValues (lib.filterAttrs (n: v: n != "all") passthru.optional-dependencies))
```

Comment (happysalada) [reply pkgs/development/python-modules/markdown2/default.nix]: I'm not a huge fan of this, but I don't really want to spend a lot of time on this PR, so I just went with it.

**Pull request merged**